carapace-server/src/main/java/org/carapaceproxy/server/RequestHandler.java

```diff
@@ -211,39 +211,44 @@ public void start() {
             case ACME_CHALLENGE:
             case REDIRECT:
                 return;
-            case PROXY:
+            case PROXY: {
+                EndpointConnection connection;
                 try {
 //                    LOG.log(Level.SEVERE, "TIME"+TIME_TRACKER.incrementAndGet()+" start " + this + " thread " + Thread.currentThread().getName());
-                    EndpointConnection connection = connectionToClient.connectionsManager.getConnection(new EndpointKey(action.host, action.port));
-                    connectionToEndpoint.set(connection);
-                    connection.sendRequest(request, this);
+                    connection = connectionToClient.connectionsManager.getConnection(new EndpointKey(action.host, action.port));                    
```

Comment:
I am cleaning up the try-catch block, EndpointNotAvailableException can be thrown only by "getConnection"